### PR TITLE
added completion spec for hardhat

### PR DESCRIPTION
**Pull request by Nutlope**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
add completion spec for hardhat
**What is the current behavior? (You can also link to an open issue here)**
no completion spec for hardhat
**What is the new behavior (if this is a feature change)?**
new completion spec for hardhat
**Additional info:**
https://hardhat.org/hardhat-network/

### CODE REVIEW

File: dev/hardhat.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      description: "Flattens and prints contracts and their dependencies",
+    },
+    { name: "help", description: "Prints this message" },
+    {
+      name: "node",
+      description: "Starts a JSON-RPC server on top of Hardhat Network",
+      options: [
+        {
+          name: "--fork",
+          description: "The URL of the JSON-RPC server to fork from",
```
Comment (Monkeyank) [inline dev/hardhat.ts:60]: Seems like this takes an arg

File: dev/hardhat.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        {
+          name: "--fork-block-number",
+          description: "The block number to fork from",
```
Comment (Monkeyank) [inline dev/hardhat.ts:64]: Seems like this takes an arg

File: dev/hardhat.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        {
+          name: "--hostname",
+          description: "The host to which to bind to for new connections",
```
Comment (Monkeyank) [inline dev/hardhat.ts:68]: Seems like this takes an arg

File: dev/hardhat.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        {
+          name: "--port",
+          description:
+            "The port on which to listen for new connections (default: 8545)",
```
Comment (Monkeyank) [inline dev/hardhat.ts:73]: Seems like this takes an arg

File: package-lock.json [hunk 5]
```diff
@@ -1,8 +1,5576 @@
 {
   "name": "linter",
   "version": "1.0.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
   "requires": true,
+  "packages": {
```
Comment (Monkeyank) [inline package-lock.json:6]: Are these changes on purpose?

**Review by Monkeyank — CHANGES_REQUESTED** [review]

File: dev/hardhat.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
+export const completionSpec: Fig.Spec = {
+  name: "hardhat",
+  options: [
+    { name: "config", description: "A Hardhat config file" },
```
Comment (QuiiBz) [inline dev/hardhat.ts:4]: Looks like this option takes an argument with a template of `filepaths`

File: dev/hardhat.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    { name: "--emoji", description: "Use emoji in messages" },
+    {
+      name: "--max-memory",
+      description: "The maximum amount of memory that Hardhat can use",
```
Comment (QuiiBz) [inline dev/hardhat.ts:8]: Looks like this option takes an argument

File: dev/hardhat.ts [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    },
+    {
+      name: "--help",
+      description: "Shows the help text or task's help if name is provided",
+      args: { isOptional: true, name: "task" },
+    },
+    { name: "--network", description: "The network to connect to" },
```
Comment (QuiiBz) [inline dev/hardhat.ts:15]: Same as above

**Review by QuiiBz — CHANGES_REQUESTED** [review]

**Comment by Monkeyank** [comment]
Also use the `export default` syntax from now on

**Pull request merged**